### DISCUSSION TITLE
MQTT retain info, state and status messages

### DISCUSSION THREAD

**Discussion by jeroenst** [arendst/Tasmota#11067] [Ideas]
I try to switch from my own software ( https://github.com/jeroenst/espmqtt ) to tasmota.

In my software all items are retained except for button press events.

By using retain in the MQTT server I always have the latest data present when a client connects to the MQTT broker.

This is very handy for a lot of things. I have for example a status page which connects to the MQTT server with javascript and requests al information (like uptime, rssi, ip address, software version etc.) from the MQTT broker and displays them right away.

WIth the (for example STATE) messages not being retained it takes a long time before the status page is filled with all the data which is not an option for me.

I could create a PHP process to gather all the data and republish them retained in an mqtt topic but that is more a workaround for me.

I would like to make the tele/INFOx and tele/STATE topic retained.

Is it possible to implement this?

**Comment by barbudor** [comment]
You can look at the list of commands for commands such as PowerRetain, SensorRetain, etc...

**Reply by jeroenst** [reply]
There is no such thing as stateretain and inforetain afaik.

**Reply by barbudor** [reply]
You can propose a PR for InfoRetain and StateRetain. Good opport to dive in Tasmota 👍

**Reply by jeroenst** [reply]
done. https://github.com/arendst/Tasmota/pull/11084

**Reply by barbudor** [reply]
Congrats 👍

**Comment by jeroenst** [comment]
I forgot status retain which is related, will make a PR for that.

**Comment by jeroenst** [comment]
statusretain added https://github.com/arendst/Tasmota/pull/11109

Status can be triggered on mqtt connect with this rule:

```
Rule1
  on Mqtt#Connected do Status 0 endon

Rule1 1
```
